nautobot/extras/management/__init__.py

```diff
@@ -153,24 +229,35 @@ def create_custom_statuses(
         # Do not print output during unit testing migrations
         verbosity = 1
 
-    if not models:
-        models = CHOICESET_MAP.keys()
-
-    # Prep the app and get the Status model dynamically
+    choiceset_map = {}
+    if metadatamodel.lower() == "role":
+        choiceset_map = ROLE_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    elif metadatamodel.lower() == "status":
+        choiceset_map = STATUS_CHOICESET_MAP
+        if not models:
+            models = choiceset_map.keys()
+    else:
+        raise ValueError("only status and role are supported for _create_custom_role_or_status_instances")
+
+    # Prep the app and get the model dynamically
     try:
-        Status = apps.get_model("extras.Status")
+        MetadatModel = apps.get_model(f"extras.{metadatamodel}")
         ContentType = apps.get_model("contenttypes.ContentType")
     except LookupError:
-        return
+        raise LookupError(
+            f"Could not find extras.{metadatamodel} and/or contenttypes.ContentType. Please make sure the correct migration dependencies are set before using this method"
+        )
```

Comment:
Just confirming that you've tested forward and reverse migrations with this change in behavior?

Reply:
It was kinda difficult to test this behavior because Im running into issues when unapplying `contenttypes 0001` and/or  `extras 0001`. But [this](https://docs.djangoproject.com/en/5.0/topics/migrations/#accessing-models-from-other-apps) should confirm that this change in behavior should be correct

Reply:
I'm okay if you can't find a scenario that triggers this exception. Just wanted to make sure that forward/reverse migrations are *not* triggering the exception. :-)